src/spi_async.rs

```diff
@@ -0,0 +1,554 @@
+use crate::interrupt::{IsrCriticalSection, IsrCriticalSectionGuard};
+use crate::spi::{config, to_spi_err, Lock, SpiDriver, SpiError};
+use core::borrow::BorrowMut;
+use core::cell::UnsafeCell;
+use core::cmp::max;
+use core::cmp::Ordering;
+use core::future::Future;
+use core::iter::Zip;
+use core::marker::PhantomPinned;
+use core::ops::{Deref, DerefMut};
+use core::pin::Pin;
+use core::slice::Chunks;
+use core::slice::ChunksMut;
+use core::task::{Context, Poll, Waker};
+use embedded_hal::spi::ErrorType;
+use embedded_hal_async::spi::{SpiBus, SpiBusFlush, SpiBusRead, SpiBusWrite};
+use esp_idf_sys::*;
+
+pub struct SpiBusDriver<T> {
+    _driver: T,
+    handle: spi_device_handle_t,
+    _lock: Lock,
+    trans_len: usize,
+}
+
+// TODO: IRAM
```

Comment:
I don't think IRAM is necessary (fortunately). Otherwise, that would be a total PITA. :)

Reply:
Yeah I wasn't sure how we'd get the standard library to put the `Waker` methods in IRAM as well. Good things we don't have to.

Reply:
It is not just the `Waker` methods. it is also *what the waker wakes* (as in portions of the executor code, that (re)schedules the task being woken). It becomes so hairy, that IMO the IRAM option is just not practical in most of the async use cases. Hence this request:
https://github.com/espressif/esp-idf/issues/10250